src/test/java/com/stripe/StripeMockProcess.java

```diff
@@ -0,0 +1,153 @@
+package com.stripe;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Field;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.Getter;
+
+public class StripeMockProcess {
+
+  private static final String ROOT = "src/test";
+  private static final String FIXTURE_PATH = "/openapi/fixtures3.json";
+  private static final String SPEC_PATH = "/openapi/spec3.json";
+
+  private static Process process;
+  @Getter
+  private static String port;
+  private static String pid;
+
+  /**
+   * Start embedded stripe-mock when spec and fixture resources are detected.
+   *
+   * @return whether the stripe-mock can be started and is now running.
+   */
+  public static boolean start() throws IOException, InterruptedException, NoSuchFieldException,
+      IllegalAccessException {
+
+    if (!new File(getPathSpec()).exists() || !new File(getPathFixture()).exists()) {
+      return false;
+    }
+
+    if (process != null && isAlive(process)) {
+      System.out.println(String.format("stripe-mock is already running, port = %s", port));
+      return true;
+    } else {
+      // to reinitialize all properties
+      process = null;
+      port = null;
+      pid = null;
+    }
+
+    ProcessBuilder processBuilder = new ProcessBuilder()
+        .command(
+            "stripe-mock",
+            "-http-port",
+            "0", // stripe-mock to choose an available port
+            "-spec",
+            getPathSpec(),
+            "-fixtures",
+            getPathFixture()
+        );
+
+    try {
+      process = processBuilder.start();
+    } catch (IOException e) {
+      System.out.println(String.format(
+          "Error while starting stripe-mock, port = %d, fixtures = %s and %s, "
+              + "error message = %s",
+          port, getPathSpec(), getPathFixture(), e.getMessage()
+      ));
+      System.exit(1);
+    }
+
+    pid = getProcessId(process);
+
+    if (isAlive(process)) {
+      System.out.println(String.format("Started stripe-mock, PID = %s", pid));
+      System.out.println("Finding port bound to stripe-mock...");
+      port = detectBoundPort(process);
```

Comment:
This would block if there's no process output. 
I could potentially make this non-blocking as a thread, and continually ping within the timeout.
Given the approach of output string pattern matching, we already assume the predictable pattern of our internal tooling, maybe it's not worthwhile to implement the time-out pattern catching when there's no output.
I also put this block after checking that stripe-mock is alive, to reduce chance of hanging when somehow the process exits.

Alternatively, I thought about using `lsof -Pan -p <stripe mock PID> -i` and get the response. This can make our port finding independent of stripe-mock implementation, but the downside is perhaps how we have variance to lsof string output which makes pattern matching more fragile. Mine looks something like this.
```
COMMAND     PID             USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
stripe-mo 20288 mickjermsurawong    3u  IPv6 0x4dxxxxx      0t0  TCP *:51516 (LISTEN)
```

Reply:
I believe `readPartial` in ruby's implementation will have the same problem on "It blocks only if ios has no data immediately available." [doc](https://ruby-doc.org/core-2.2.2/IO.html#method-i-readpartial)

Reply:
Nice, thanks for digging into this Mick. I think we can just leave it at this. There's really no reason for the stripe-mock process to be alive and not output anything, and if that ever happens we can just look into it.